### PR TITLE
Passing default_transformation options while showing media library

### PR DESCRIPTION
**Pull request by amitkhanal**
**Summary**
This initializes the Cloudinary Media Library with default_transformations during the show() function invocation.

**Motivation**
default_transformation option wasn't passed when rendering the media library. As a result, transformations didn't work properly.

### CODE REVIEW

File: packages/netlify-cms-media-library-cloudinary/src/index.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ async function init({ options = {}, handleInsert } = {}) {
   const resolvedOptions = { ...defaultOptions, ...integrationOptions };
   const cloudinaryConfig = { ...defaultConfig, ...providedConfig, ...enforcedConfig };
   const cloudinaryBehaviorConfigKeys = ['default_transformations', 'max_files', 'multiple'];
-  const cloudinaryBehaviorConfig = pick(cloudinaryConfig, cloudinaryBehaviorConfigKeys);
+  const cloudinaryBehaviorConfig = pick(resolvedOptions, cloudinaryBehaviorConfigKeys);
```
Comment (erquhart) [inline packages/netlify-cms-media-library-cloudinary/src/index.js]: `resolvedOptions` is made up of keys that are internal to Netlify CMS, like `use_secure_url`. `cloudinaryConfig` has the value of `media_library.config` from the CMS configuration, which contains Cloudinary specific keys exclusively.

File: packages/netlify-cms-media-library-cloudinary/src/index.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ async function init({ options = {}, handleInsert } = {}) {
       if (allowMultiple === false) {
         instanceConfig.multiple = false;
       }
-      return mediaLibrary.show({ config: { ...cloudinaryBehaviorConfig, ...instanceConfig } });
```
Comment (erquhart) [inline packages/netlify-cms-media-library-cloudinary/src/index.js]: We added `config` on the advice of your team, did the API change?

**Comment by erquhart** [comment]
Thanks for opening this @amitkhanal!

It's possible I'm missing something, as working late on a weekend often entails, but I don't know how the changes here would have resulted in a working Cloudinary integration. Specifics are in my comments. If you're certain this worked for you in testing I'll pull it down and test it out on Monday.

File: packages/netlify-cms-media-library-cloudinary/src/index.js (hunk 1 above)
Comment (amitkhanal) [reply packages/netlify-cms-media-library-cloudinary/src/index.js]: `resolvedOptions` contains `defaultOptions` and `integrationOptions ` (line 54). It is used to create `cloudinaryBehaviorConfig` which contains Cloudinary related configurations defined under `cloudinaryBehaviorConfigKeys`. 

`cloudinaryConfig` contains `api_key` and `cloud_name` and only needs to be passed when creating the media library. The `mediaLibrary.show` function  requires `default_transformations` which is now passed through `resolved_options`. Without this, the default transformations were not applied to the inserted image.

File: packages/netlify-cms-media-library-cloudinary/src/index.js (hunk 2 above)
Comment (amitkhanal) [reply packages/netlify-cms-media-library-cloudinary/src/index.js]: I will double check with the team but adding `config` does not set the configurations for media library.

**Comment by amitkhanal** [comment]
Hey @erquhart, thanks for your comments. I've replied with mine. Please review.

There were a couple of issues - 
1. `default_transformation` was not being set in Media Library and as a result the insert images didn't have transformations.
2. The `default_transformation` needs to be of type `Object[][]`. You can check the type at - https://cloudinary.com/documentation/media_library_widget#show_or_hide_the_media_library.

The example shown was incorrect. That is being fixed.

File: packages/netlify-cms-media-library-cloudinary/src/index.js (hunk 1 above)
Comment (erquhart) [reply packages/netlify-cms-media-library-cloudinary/src/index.js]: > `cloudinaryConfig` contains `api_key` and `cloud_name` and only needs to be passed when creating the media library.

`cloudinaryConfig` contains all configuration for the cloudinary media player. This line of code isn't passing `cloudinaryConfig`, it's using `lodash.pick` to select the behavior config keys from it.

Comment (erquhart) [reply packages/netlify-cms-media-library-cloudinary/src/index.js]: This is a screenshot from the console while running the branch from this PR. Notice the keys present in `resolvedConfig` - they're keys invented for the Netlify CMS integration, not anything that the Cloudinary widget would be able to use.

screen shot 2019-02-06 at 10 22 47 am

**Comment by erquhart** [comment]
Just checked with the current CMS, default transformations do work:

1. Hit the testbed codepen: https://codepen.io/erquhart/pen/PVOopW
2. In the JS panel, find the CMS config and add your cloud name and api key
3. Create a new post and insert an image

You'll see the inserted image is pixelated via `default_transformations` in the config. As Nadav mentioned, `config` is deprecated, so removing that key in the `show` config is good 👍 

I do see that not all transformations work on Cloudinary's end, eg. `width` just doesn't work no matter what, though it's being passed the same as other transformations that do work.

**Comment by erquhart** [comment]
Also seeing that `default_transformations` passed to `show` don't seem to be picked up by the Cloudinary widget. The `default_transformations` passed in during initialization are respected, though. This is based on parsing the data object received from the widget, before Netlify CMS touches it.

Any chance the API is somehow different for `show` than for `createMediaLibrary`?

**Comment by amitkhanal** [comment]
> eg. `width` just doesn't work no matter what, though it's being passed the same as other transformations that do work.

`width` requires `crop` mode be the provided as well. The sample configuration at https://www.netlifycms.org/docs/cloudinary/#global-configuration has `crop` along with `width` transformation. I will bring this up internally to add this on our documentation. 
Besides `width`, are there other transformations that you do not see working?

File: packages/netlify-cms-media-library-cloudinary/src/index.js (hunk 1 above)
Comment (amitkhanal) [reply packages/netlify-cms-media-library-cloudinary/src/index.js]: Did you have the `default_transformation` defined in the config? Here's what I have for `resolvedOptions`
screen shot 2019-02-08 at 4 15 29 pm

**Comment by erquhart** [comment]
I thought there were others, but blanking right now.

The docs shows `width` as a resizing option, and don't mention cropping: https://cloudinary.com/documentation/image_transformation_reference#width_parameter

I'd expect setting width to allow an entire image to be resized based on that, am I missing something?

**Comment by amitkhanal** [comment]
> Also seeing that `default_transformations` passed to `show` don't seem to be picked up by the Cloudinary widget. The `default_transformations` passed in during initialization are respected, though. This is based on parsing the data object received from the widget, before Netlify CMS touches it.
> 
> Any chance the API is somehow different for `show` than for `createMediaLibrary`?



> I thought there were others, but blanking right now.
> 
> The docs shows `width` as a resizing option, and don't mention cropping: https://cloudinary.com/documentation/image_transformation_reference#width_parameter
> 
> I'd expect setting width to allow an entire image to be resized based on that, am I missing something?

The `width` can be passed by itself when used in transformation URLs directly. However, they are required to be passed with `crop` mode otherwise.

**Comment by amitkhanal** [comment]
> Also seeing that `default_transformations` passed to `show` don't seem to be picked up by the Cloudinary widget. The `default_transformations` passed in during initialization are respected, though. This is based on parsing the data object received from the widget, before Netlify CMS touches it.
> 
> Any chance the API is somehow different for `show` than for `createMediaLibrary`?

Passing `default_transformations` on show method works. Please check - https://jsfiddle.net/amitkhanal/asnjo59h/77/

I've also verified this in CMS with `field` level configuration.

**Comment by amitkhanal** [comment]
@erquhart I've reverted the change and it's now using 'cloudinaryConfig' instead of 'resolvedOptions'. Please review.

**Comment by erquhart** [comment]
@amitkhanal looks good, just need to get tests passing for merge.

**Comment by amitkhanal** [comment]
@erquhart I've fixed formatting and also updated the tess. Please verify,

**Pull request merged**